src/codeu/chat/common/DerbyDatabaseInteractions.java

```diff
@@ -0,0 +1,39 @@
+package codeu.chat.common;
+
+import codeu.chat.util.store.Store;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import codeu.chat.util.Uuid;
+
+public interface DerbyDatabaseInteractions {
+	
+	// USER
+	public Store<Uuid, User> getAllUsers();
+	
+	public User getUser(Uuid userid) throws SQLException, IOException;
```

Comment:
I wonder if the interface functions need to throw SQLExceptions. What if I want to create an implementation of this interface that doesn't use a database, and therefore will never throw a SQLException? (For that matter, since the implementation uses predefined SQL, when would it ever throw a SQLException at all?)